src/main/java/wedlog/address/logic/parser/GuestEditCommandParser.java

```diff
@@ -0,0 +1,100 @@
+package wedlog.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static wedlog.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static wedlog.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static wedlog.address.logic.parser.CliSyntax.PREFIX_DIETARY;
+import static wedlog.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static wedlog.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static wedlog.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static wedlog.address.logic.parser.CliSyntax.PREFIX_RSVP;
+import static wedlog.address.logic.parser.CliSyntax.PREFIX_TABLE;
+import static wedlog.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+
+import wedlog.address.commons.core.index.Index;
+import wedlog.address.logic.commands.GuestEditCommand;
+import wedlog.address.logic.commands.GuestEditCommand.EditGuestDescriptor;
+import wedlog.address.logic.parser.exceptions.ParseException;
+import wedlog.address.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new GuestEditCommand object
+ */
+public class GuestEditCommandParser implements Parser<GuestEditCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the GuestEditCommand
+     * and returns an GuestEditCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public GuestEditCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
+                        PREFIX_RSVP, PREFIX_DIETARY, PREFIX_TABLE, PREFIX_TAG);
+
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, GuestEditCommand.MESSAGE_USAGE), pe);
+        }
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
+                PREFIX_RSVP, PREFIX_DIETARY, PREFIX_TABLE);
+
+        EditGuestDescriptor editGuestDescriptor = new EditGuestDescriptor();
+
+        // TODO: UPDATE AFTER SAMUEL'S PR IS MERGED
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            editGuestDescriptor.setName(ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get()));
+        }
+        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+            editGuestDescriptor.setPhone(ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get()));
```

Comment:
Your current implementation assumes all fields are compulsory. Perhaps you could make use of the new method `parseIfNotBlank`?

```suggestion
    editGuestDescriptor.setPhone(ParserUtil.parseIfNotBlank(argMultimap.getValue(PREFIX_PHONE).get(),
                    ParserUtil::parsePhone));
```